### ISSUE TITLE
qasync with QtQuick and PySide6

### ISSUE THREAD

**Issue by liquid-metal** [CabbageDevelopment/qasync#50]
Hello, not an issue technically (probably), but I am absolutely stuck.
I am trying to use qasync with QtQuick in PySide6. In the following test application, the seems not to be called (though the connection does work, as passing an argument from QML leads to an asyncio error). Substituting asyncSlot with Slot from PySide makes the example work.
Can you please give me a hint?

main.py
```import sys
import qasync
from PySide6.QtCore import QObject
from PySide6.QtGui import QGuiApplication
from PySide6.QtQml import QQmlApplicationEngine, qmlRegisterType


class TestController(QObject):
    @qasync.asyncSlot()
    async def test_slot(self):
        print('test async slot')


def master():
    app = QGuiApplication(sys.argv)
    engine = QQmlApplicationEngine()

    qmlRegisterType(TestController, 'test', 1, 0, 'TestController')

    engine.load('main.qml')

    sys.exit(app.exec())


if __name__ == '__main__':
    qasync.run(master())
```
main.qml
```import QtQuick
import QtQuick.Controls
import QtQuick.Window

import test

ApplicationWindow {
    id: mainWindow
    visible: true

    Rectangle {
        anchors.fill: parent
        color: '#00ff00'

        MouseArea {
            anchors.fill: parent

            onClicked: testController.test_slot()
        }
    }

    TestController {
        id: testController
    }
}
```

**Comment by eyllanesc** [comment]
@liquid-metal  You are not using qasync correctly.

```python
import asyncio
import sys

from PySide6.QtCore import QObject, Signal
from PySide6.QtGui import QGuiApplication
from PySide6.QtQml import QQmlApplicationEngine, qmlRegisterType

import qasync


class TestController(QObject):
    started = Signal()
    valueChanged = Signal(int)
    finished = Signal()

    @qasync.asyncSlot(int)
    async def test_slot(self, end_value):
        self.started.emit()
        for i in range(end_value):
            await asyncio.sleep(1)
            print(f"python: {i}")
            self.valueChanged.emit(i)
        self.finished.emit()


def main():
    app = QGuiApplication(sys.argv)

    loop = qasync.QEventLoop(app)
    engine = QQmlApplicationEngine()

    qmlRegisterType(TestController, "test", 1, 0, "TestController")

    engine.load("main.qml")

    with loop:
        loop.run_forever()


if __name__ == "__main__":
    main()
```

```qml
import QtQuick.Controls
import QtQuick.Window
import test

ApplicationWindow {
    id: mainWindow

    visible: true

    Rectangle {
        anchors.fill: parent
        color: '#00ff00'

        MouseArea {
            anchors.fill: parent
            onClicked: testController.test_slot(5)
        }

    }

    TestController {
        id: testController

        onStarted: function() {
            console.log("started");
        }
        onFinished: function() {
            console.log("finished");
        }
        onValueChanged: function(i) {
            console.log(i);
        }
    }

}
```

**Comment by skaslev** [comment]
I think this can be closed with #53 merged.
